### PR TITLE
Updating no-side-effects rule to better detect sets inside of blocks

### LINKED ISSUE #259: Side effects in conditions are not detected
The next code throws a no-side-effects error:

```
foo: computed(function() {
  this.set('bar', 'someValue');
})
```

Yet this code doesn't:

```
foo: computed(function() {
  const condition = true;
  if (condition) {
    this.set('bar', 'someValue');
  }
})
```

### PR DESCRIPTION
**Pull request by gmurphey**
Fixes #259.

### CODE REVIEW

**Comment by gmurphey** [comment]
@rwjblue those test cases have been added and the rules updated to make them pass. Thanks!

File: lib/rules/no-side-effects.js [hunk 1]
```diff
@@ -18,42 +19,42 @@ module.exports = {
...
+
     const message = 'Don\'t introduce side-effects in computed properties';
 
     const report = function (node) {
       context.report(node, message);
     };
 
     return {
+      ImportDeclaration(node) {
+        emberImportAliasName = ember.getEmberImportAliasName(node);
```
Comment (rwjblue) [inline lib/rules/no-side-effects.js]: I don't think this is quite correct. A given module will have _many_ import statements, with the logic here we are clobbering `emberImportAliasName` for each one.

Can you add the following good test:

```js
import Ember from 'ember';
import Foo from "some-other-thing"; 
let foo = computed("test", function() { 
  Foo.set(this, "testAmount", test.length); 
  return ""; 
});
```

**Comment by rwjblue** [comment]
@gmurphey had any time to look into the updates?

**Comment by gmurphey** [comment]
Found some time this afternoon, @rwjblue! Thanks for the feedback.

**Comment by rwjblue** [comment]
Thank you!

**Pull request merged**